pyproject.toml

```diff
@@ -0,0 +1,55 @@
+[build-system]
+requires = ["setuptools>=42.0.0", "wheel"]
+build-backend = "setuptools.build_meta"
+
+
+[tool.black]
```

Comment:
What is the black configuration in the toml for exactly? (I know `black`, but I am not sure why to put it here; I am not so familiar with the toml way of setting things up.)

Reply:
Everything in the `pyproject.toml` is copied directly from the [cookiecutter napari plugin](https://github.com/napari/cookiecutter-napari-plugin) example. So this is here because whoever set that up liked those formatting tools. We can absolutely remove this if you want.

Reply:
I realized this doesn't actually answer your question of "why put the configuration settings in the pyproject.toml file?"
The answer to that is it is potentially useful if you want configuration settings specific for this particular project (but here it's all very generic so perhaps doesn't make as much sense).
And yes, you could put the black config settings in their own separate file - I guess people prefer to put everything in the pyrpoject.toml because it can get messy with a lot of scattered config files if you use a lot of different tools together.

[Here is what the black docs have to say about when this might be useful](https://black.readthedocs.io/en/stable/usage_and_configuration/the_basics.html#configuration-via-a-file):
> Black is able to read project-specific default values for its command line options from a pyproject.toml file. This is especially useful for specifying custom --include and --exclude/--force-exclude/--extend-exclude patterns for your project.